torch/_inductor/codegen/common.py

```diff
@@ -1029,3 +1019,83 @@ class OptimizationContext:
 
     # Load uint8 value as float32
     is_load_uint8_as_float: bool = False
+
+
+@functools.lru_cache(None)
+def jinja2_env():
+    try:
+        import jinja2
+
+        return jinja2.Environment(
+            undefined=jinja2.StrictUndefined,
+        )
+    except ImportError:
+        return None
+
+
+class ChoiceCaller:
```

Comment:
A small doc comment would be great: What's the purpose of this class / which problem does it solve? Is it supposed to have subclasses?